src/systems/trajectory_follower/TrajectoryFollower.cc

```diff
@@ -0,0 +1,353 @@
+  {
+    igndbg << "Circle element activated" << std::endl;
+    auto circleElem = _sdf->GetElement("circle");
+
+    if (!circleElem->HasElement("radius"))
+    {
+      ignerr << "No <circle><radius> specified" << std::endl;
+      return;
+    }
+
+    // Parse the required <direction> field.
+    double radius = circleElem->Get<double>("radius");
+
+    // Get the current model position in global coordinates. Create
+    // local vectors that represent a path along a rough circle.
+    ignition::math::Vector2d position(this->modelPose.Pos().X(),
+                                      this->modelPose.Pos().Y());
+    double angle = 0;
+    ignition::math::Vector2d vec(radius, 0);
+    for (unsigned int i = 0u; i < this->numSamples; ++i)
+    {
+      // Add the local vector to the current position.
+      // Store global position as a waypoint.
+      this->localWaypoints.push_back(position + vec);
+      angle += 2 * M_PI / this->numSamples;
+      vec.Set(radius * cos(angle), radius * sin(angle));
+      igndbg << "Entered circle waypoint " << position + vec << std::endl;
+    }
+  }
+  // If no waypoints or circle, check for the <line> element and parse.
+  else if (_sdf->HasElement("line"))
+  {
+    auto lineElem = _sdf->GetElement("line");
+    // Parse the required <direction> field.
+    if (!lineElem->HasElement("direction"))
+    {
+      ignerr << "No <line><direction> specified" << std::endl;
+      return;
+    }
+    auto dirElem = lineElem->GetElement("direction");
+    double direction = dirElem->Get<double>();
+
+    // Parse the required <length> field.
+    if (!lineElem->HasElement("length"))
+    {
+      ignerr << "No <line><length> specified" << std::endl;
+      return;
+    }
+    auto lenElem = lineElem->GetElement("length");
+    double length = lenElem->Get<double>();
+
+    // Create a relative vector in the direction of "direction" and of
+    // length "length".
+    ignition::math::Vector3d lineVec(
+      length * cos(direction * M_PI / 180),
+      length * sin(direction * M_PI / 180), 0);
+    ignition::math::Vector2d position(this->modelPose.Pos().X(),
+                                      this->modelPose.Pos().Y());
+    // Add the initial model position and calculated endpoint as waypoints.
+    this->localWaypoints.push_back(position);
+    ignition::math::Vector3d p = this->modelPose.CoordPositionAdd(lineVec);
+    ignition::math::Vector2d p2D = {p.X(), p.Y()};
+    this->localWaypoints.push_back(p2D);
+    igndbg << "Entered line waypoints " << position << ", " << p2D << std::endl;
+  }
+
+  // Parse the optional <loop_forever> element.
+  if (_sdf->HasElement("loop_forever"))
+    this->loopForever = _sdf->Get<bool>("loop_forever");;
+
+  // Parse the optional <force> element.
+  if (_sdf->HasElement("force"))
+    this->forceToApply = _sdf->Get<double>("force");
+
+  // Parse the optional <torque> element.
+  if (_sdf->HasElement("torque"))
+    this->forceToApply = _sdf->Get<double>("torque");
```

Comment:
`forceToApply` -> `torqueToApply`

Reply:
after fixing this, I made both `<force>` and `<torque>` in demo world file to be `10` and the box moves nicely without too much overshooting when rotating

Reply:
Good catch, changed in [f082490](https://github.com/ignitionrobotics/ign-gazebo/pull/1332/commits/f082490540e4e27aa58065479729b5605044bec7).